### ISSUE TITLE
RuntimeWarning

### ISSUE THREAD

**Issue by bukosabino** [bukosabino/ta#83]
```sh
trend.py:170: RuntimeWarning: invalid value encountered in double_scalars
  dip[i] = 100 * (dip_mio[i]/trs[i])

trend.py:174: RuntimeWarning: invalid value encountered in double_scalars
  din[i] = 100 * (din_mio[i]/trs[i])
```

**Comment by AntMatRod** [comment]
site-packages\ta\trend.py:468: RuntimeWarning: invalid value encountered in double_scalars
  dip[i] = 100 * (self._dip[i]/self._trs[i])

site-packages\ta\trend.py:472: RuntimeWarning: invalid value encountered in double_scalars
  din[i] = 100 * (self._din[i]/self._trs[i])

**Comment by AntMatRod** [comment]
> site-packages\ta\trend.py:468: RuntimeWarning: invalid value encountered in double_scalars
> dip[i] = 100 * (self._dip[i]/self._trs[i])
> 
> site-packages\ta\trend.py:472: RuntimeWarning: invalid value encountered in double_scalars
> din[i] = 100 * (self._din[i]/self._trs[i])

same issues

**Comment by alim7676** [comment]
same issue

**Comment by daggamault** [comment]
same issue

**Comment by msarm** [comment]
Same issue 

```
\.venv\lib\site-packages\ta\trend.py:543: RuntimeWarning: invalid value encountered in double_scalars
  dip[i] = 100 * (self._dip[i]/self._trs[i])
\.venv\lib\site-packages\ta\trend.py:547: RuntimeWarning: invalid value encountered in double_scalars
  din[i] = 100 * (self._din[i]/self._trs[i])
```

**Comment by bukosabino** [comment]
Hi all,

This is a warning error.

I will fix it asap.

Best,
Dario

**Comment by saltyDan716** [comment]
Hi, first off thanks for doing this! Second, im just curious if there is and ETA for this fix?
     Thanks again, 
     Dan

**Comment by sagarkhale** [comment]
RuntimeWarning: invalid value encountered in double_scalars
  dip[i] = 100 * (self._dip[i]/self._trs[i])

**Comment by ElliotVilhelm** [comment]
Still present

**Comment by rufusroger** [comment]
Any update on this? I am still encountering the same issue.

**Comment by RRRddd10001** [comment]
RuntimeWarning: invalid value encountered in double_scalars
  dip[i] = 100 * (self._dip[i] / self._trs[i])

this problem is still present

**Comment by karenjh2** [comment]
I get error RuntimeWarning: invalid value encountered in double_scalars
  dip[i] = 100 * (self._dip[i] / self._trs[i])

Looks like divide by zero error. I see it has been previously reported. Several times. I appreciate your library but this makes it unusable for me. If I fix it, I run into another divide by zero error. If I fix that, I run into a different error. Cascading issues.

**Comment by elginbeloy** [comment]
Yeah, this fails with pretty much all data. Even basic yahoo finance data can't run because of the divide by zero errors. Great library otherwise, though.

**Comment by YukaLangbuana** [comment]
Hey what's the timeline for this? It's not something major but it's nice to have the log clean😛

**Comment by neothe0** [comment]
Getting the same error...

**Comment by Neogus** [comment]
This is so annoying I had answer this issue with a beatiful explanation and Github decided to erase what i was writing and send me to another page gr.. So in short now, line 715 in Trend.py:

pdm = _get_min_max(self._high, close_shift, "max")
pdn = _get_min_max(self._high, close_shift, "max")
diff_directional_movement = pdm - pdn

And then goes to line 726:

 for i in range(1, len(self._trs) - 1):
            self._trs[i] = (
                self._trs[i - 1]
                - (self._trs[i - 1] / float(self._window))
                + diff_directional_movement[self._window + i]
            )

If the Dataset has missing values (NaN) then since self._trs[i] is being filled with self._trs[i-1] as part of the formula the first Nan it encounters will continue to fill all values with NaN. So you end up with a self._trs that has a lot of missing values so when you divide something with self._trs you get that warning, but of course is not only a warning you are actually messing up with the results of the indicators involved. I solved my particular issue by adding a fillna method just after line 717 like this:

pdm = _get_min_max(self._high, close_shift, "max")
pdn = _get_min_max(self._high, close_shift, "max")
diff_directional_movement = pdm - pdn
---> diff_directional_movement.fillna(method='ffill',inplace=True) <--- New line

However you can also add some vectorized numpy solution to the _get_min_max function in utils.py so it makes sure to clean all Nan values of the series that are feed to it, or maybe just fillna the whole dataset somehow.

**Comment by rundef** [comment]
@Neogus :+1: You should make a PR

**Comment by jaymegordo** [comment]
@Neogus thx that fixed it for me!

**Comment by informankur** [comment]
or we could simply add this line at the beginning of the code

`np.seterr(divide='ignore', invalid='ignore')`

We know there will be nan or 0 values at the beginning of data range while calculating an indicator, and that will throw this warning. So by adding this one line, solves that issue.

**Comment by formula-x** [comment]
> or we could simply add this line at the beginning of the code
> 
> `np.seterr(divide='ignore', invalid='ignore')`
> 
> We know there will be nan or 0 values at the beginning of data range while calculating an indicator, and that will throw this warning. So by adding this one line, solves that issue.

kinda new to this, can you explain how this solves the issue?

**Comment by informankur** [comment]
@formula-x 
The runtime warning happens when the program is trying to divide a value by zero or nan value. We know in order to calculate our Indicators, some of the initial values will be either zero or nan, which inevitably are going to throw the runtime warning. 
So, what this line does is to ignore such warnings, pertaining only to divide function. Rest of warnings will still work. 

I've tried to fix another part of the library , and similar issue cropped up, but that time I took the long method of solving it. This time I felt ignoring divide warning is a more efficient way of handling it.

**Comment by formula-x** [comment]
> @formula-x 
> 
> The runtime warning happens when the program is trying to divide a value by zero or nan value. We know in order to calculate our Indicators, some of the initial values will be either zero or nan, which inevitably are going to throw the runtime warning. 
> 
> So, what this line does is to ignore such warnings, pertaining only to divide function. Rest of warnings will still work. 
> 
> 
> 
> I've tried to fix another part of the library , and similar issue cropped up, but that time I took the long method of solving it. This time I felt ignoring divide warning is a more efficient way of handling it. 

Ok I get it, thanks for the explanation bro I appreciate it.. do you have a discord or anything I can reach you on incase I have more questions pertaining to this library? Judging by your reply, there are other areas of this library where I might run into similar issues..

**Comment by informankur** [comment]
sure, leave a comment here, will reply to you :-) This library is very well done, this is the only minor irritant.

**Comment by ziyunxiao** [comment]
Still have this error on different line:
/usr/local/lib/python3.7/dist-packages/ta/trend.py:768: RuntimeWarning: invalid value encountered in double_scalars
  dip[i] = 100 * (self._dip[i] / self._trs[i])
/usr/local/lib/python3.7/dist-packages/ta/trend.py:772: RuntimeWarning: invalid value encountered in double_scalars
  din[i] = 100 * (self._din[i] / self._trs[i])

**Comment by summa-code** [comment]
For me, it started popping up recently. Same issue.

**Comment by jonnymaserati** [comment]
You’d be better slicing out the bits where you expect the error, otherwise you might not catch genuine errors.

**Comment by jonnymaserati** [comment]
Getting some similar errors:

```terminal
/trend.py:769: RuntimeWarning: invalid value encountered in double_scalars
/trend.py:774: RuntimeWarning: invalid value encountered in double_scalars
/trend.py:938: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  self._psar_up = pd.Series(index=self._psar.index)
trend.py:939: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  self._psar_down = pd.Series(index=self._psar.index)
```
Maybe do the calcs with `numpy` and wrap `with np.errstate()`?

**Comment by DaniDD** [comment]
This Commit fix this Problem:
https://github.com/bukosabino/ta/pull/196/commits/d0cccf8e31e4cfc889b0d2cbac36e2b0dcd1dbac
But there are other line numbers

**Comment by Mysterytrade827** [comment]
C:\Users\addev\anaconda3\lib\site-packages\ta\trend.py:769: RuntimeWarning: invalid value encountered in double_scalars
  dip[idx] = 100 * (self._dip[idx] / value)
C:\Users\addev\anaconda3\lib\site-packages\ta\trend.py:774: RuntimeWarning: invalid value encountered in double_scalars
  din[idx] = 100 * (self._din[idx] / value)

**Comment by gafakaya** [comment]
C:\Users\eldem\AppData\Local\Programs\Python\Python310\lib\site-packages\ta\trend.py:780: RuntimeWarning: invalid value encountered in double_scalars
  dip[idx] = 100 * (self._dip[idx] / value)
C:\Users\eldem\AppData\Local\Programs\Python\Python310\lib\site-packages\ta\trend.py:785: RuntimeWarning: invalid value encountered in double_scalars
  din[idx] = 100 * (self._din[idx] / value)
